File: python_cpp/uPIMulator_frontend/src/util/config_loader.py
class ConfigLoader:
    def __init__(self):
        pass

    @staticmethod
    def atomic_address_width() -> int:
        return 32

    @staticmethod
    def atomic_data_width() -> int:
        return 32

    @staticmethod
    def atomic_offset() -> int:
        return 0

    @staticmethod
    def atomic_size() -> int:
        return 256

    @staticmethod
    def iram_address_width() -> int:
        return 32

    @staticmethod
    def iram_data_width() -> int:
        return 96

    @staticmethod
    def iram_offset() -> int:
        return 384 * 1024

    @staticmethod
    def iram_size() -> int:
        return 48 * 1024

    @staticmethod
    def wram_address_width() -> int:
        return 32

    @staticmethod
    def wram_data_width() -> int:
        return 32

    @staticmethod
    def wram_offset() -> int:
        return 512

    @staticmethod
    def wram_size() -> int:
        return 128 * 1024

    @staticmethod
    def stack_size() -> int:
        return 2 * 1024

    @staticmethod
    def heap_size() -> int:
        return 4 * 1024

    @staticmethod
    def mram_address_width() -> int:
        return 32

    @staticmethod
    def mram_data_width() -> int:
        return 32

    @staticmethod
    def mram_offset() -> int:
        return 512 * 1024

    @staticmethod
    def mram_size() -> int:
        return 64 * 1024 * 1024

    @staticmethod
    def num_gp_registers() -> int:
        return 24

    @staticmethod
    def max_num_tasklets() -> int:
        return 24

    @staticmethod
    def min_access_granularity() -> int:
        return 8

File: golang_vm/uPIMulator/src/misc/config_loader.go
package misc

type ConfigLoader struct {
}

func (this *ConfigLoader) Init() {
}

func (this *ConfigLoader) AddressWidth() int {
	return 32
}

func (this *ConfigLoader) AtomicDataWidth() int {
	return 32
}

func (this *ConfigLoader) AtomicOffset() int64 {
	return 0
}

func (this *ConfigLoader) AtomicSize() int64 {
	return 256
}

func (this *ConfigLoader) IramDataWidth() int {
	return 96
}

func (this *ConfigLoader) IramOffset() int64 {
	return 384 * 1024
}

func (this *ConfigLoader) IramSize() int64 {
	return 48 * 1024
}

func (this *ConfigLoader) WramDataWidth() int {
	return 32
}

func (this *ConfigLoader) WramOffset() int64 {
	return 512
}

func (this *ConfigLoader) WramSize() int64 {
	return 128 * 1024
}

func (this *ConfigLoader) MramDataWidth() int {
	return 32
}

func (this *ConfigLoader) MramOffset() int64 {
	return 512 * 1024
}

func (this *ConfigLoader) MramSize() int64 {
	return 64 * 1024 * 1024
}

func (this *ConfigLoader) StackSize() int64 {
	return 2 * 1024
}

func (this *ConfigLoader) HeapSize() int64 {
	return 4 * 1024
}

func (this *ConfigLoader) NumGpRegisters() int {
	return 24
}

func (this *ConfigLoader) MaxNumTasklets() int {
	return 24
}

func (this *ConfigLoader) VmBankOffset() int64 {
	return 512
}

func (this *ConfigLoader) VmBankSize() int64 {
	return 128 * 1024 * 1024
}

func (this *ConfigLoader) VmBg0() int {
	return 6
}

func (this *ConfigLoader) VmBg1() int {
	return 17
}

func (this *ConfigLoader) VmBank() int {
	return 18
}

func (this *ConfigLoader) VmMemorySize() int64 { return 1024 }

func (this *ConfigLoader) GarbageCollectionThreshold() int64 {
	return 100
}
